code-analysis/crates/codegen/parser/src/chumsky/typescript_lib_boilerplate.rs

```diff
@@ -63,19 +63,17 @@ pub fn lex_head() -> TokenStream {
 
             #[napi(getter)]
             pub fn start(&self) -> usize {
-                if let Node::Chars(range) = self.0.as_ref() {
-                    range.start
-                } else {
-                    unreachable!("Expected a chars node")
+                match self.0.as_ref() {
+                    Node::Chars(range) => range.start,
+                    _  => unreachable!()
                 }
             }
 
             #[napi(getter)]
             pub fn end(&self) -> usize {
-                if let Node::Chars(range) = self.0.as_ref() {
-                    range.end
-                } else {
-                    unreachable!("Expected a chars node")
+                match self.0.as_ref() {
```

Comment:
Perhaps unrelated to this PR, but thoughts on removing `unreachable!()`, `panic!()`, `unwrap()` and other panic-inducing utilities from the codebase?

I have been gradually replacing them with `anyhow::Result<>` and `anyhow::Context` over the crates I'm working on, and letting all errors propagate all the way up to `build.rs` `main()`. Any build/codegen errors give me a nice trace of what exactly happened, I can often root cause it just from the error string:

```
XXX panicked at YYY. Caused by:

[1] calling code
[2] calling code
[3] root calling code....
```

Might be overkill to spend time to explicitly do this now for existing code, but should we be aiming for this over time?